File: appPickerView_1/appPickerView_1/ViewController.m
//
//  ViewController.m
//  appPickerView_1
//
//  Created by Guest RETBOT on 30/09/22.
//

#import "ViewController.h"

@interface ViewController (){
    NSArray *paises;
}

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view.
    
    paises = @[@"Australia", @"China", @"Francia", @"Inglaterra", @"Japon", @"Mexico"];

    _tipoDeCambio =[[NSArray alloc] initWithObjects:
                    [NSNumber numberWithFloat:0.9922],
                    [NSNumber numberWithFloat:6.5938],
                    [NSNumber numberWithFloat:0.7270],
                    [NSNumber numberWithFloat:0.6206],
                    [NSNumber numberWithFloat:81.57],
                    [NSNumber numberWithFloat:20.08],nil];
    self.pickerView1.dataSource = self;
    self.pickerView1.delegate = self;
    _lblResultado.text = paises[0];
}

-(NSInteger) pickerView:(UIPickerView *)pickerView numberOfRowsInComponent:(NSInteger)component{
    return [paises count];
}

-(NSString *)pickerView:(UIPickerView *)pickerView titleForRow:(NSInteger)row forComponent:(NSInteger)component{
    return paises[row];
}

- (void)pickerView:(UIPickerView *)pickerView didSelectRow:(NSInteger)row inComponent:(NSInteger)component{
    float valorMoneda = [[_tipoDeCambio objectAtIndex:row] floatValue];
    float dolares = [_texto1.text floatValue];
    float resultado = dolares * valorMoneda;
    
    NSString *resultadoString = [[NSString alloc] initWithFormat: @"%.2f USD = %2f %@", dolares, resultado, [paises objectAtIndex: row]];
    
    _lblResultado.text = resultadoString;
}


- (IBAction)texto1:(UITextField *)sender {
}

@end

File: APPsIOS25nov/appPicker1/appPicker1/ViewController.swift
//
//  ViewController.swift
//  appPicker1
//
//  Created by Usuario invitado on 25/11/22.
//

import UIKit

class ViewController: UIViewController, UIPickerViewDelegate , UIPickerViewDataSource{
    
    let carroColor = [["VW","Ford","Toyota","Honda","Peugot"],["Azul 🚙","Rojo 🚗", "Verde 🔫"]]

    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view.
        
  //      let pickerView = UIPickerView()
   //     pickerView.delegate = self
        pickerView1.dataSource = self
        pickerView1.delegate = self
        //texto1.inputView = pickerView
      //  texto1.inputView = pickerView1
    }
    @IBOutlet weak var texto1: UITextField!
    
    func numberOfComponents(in pickerView: UIPickerView) -> Int {
            return 2;
    }
    func pickerView(_ pickerView: UIPickerView, numberOfRowsInComponent component: Int) -> Int {
        return carroColor[component].count
    }
    
    func pickerView(_ pickerView: UIPickerView, titleForRow row: Int, forComponent component: Int) -> String? {
        return self.carroColor[component][row]
    }
    func pickerView(_ pickerView: UIPickerView, didSelectRow row: Int, inComponent component: Int) {
        
        let carro = carroColor[0][pickerView.selectedRow(inComponent: 0)]
        let color = carroColor[1][pickerView.selectedRow(inComponent: 1)]
        
        texto1.text = carro + " " + color
    }
    
    @IBOutlet weak var pickerView1: UIPickerView!
}

